acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -588,15 +608,19 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> ACVM<'a, F, B> {
         else {
             unreachable!("Not executing a Call opcode");
         };
+
+        let opcode_location =
+            ErrorLocation::Resolved(OpcodeLocation::Acir(self.instruction_pointer()));
         if *id == AcirFunctionId(0) {
-            return Err(OpcodeResolutionError::AcirMainCallAttempted {
-                opcode_location: ErrorLocation::Resolved(OpcodeLocation::Acir(
-                    self.instruction_pointer(),
-                )),
-            });
+            return Err(OpcodeResolutionError::AcirMainCallAttempted { opcode_location });
         }
 
-        if is_predicate_false(&self.witness_map, predicate)? {
+        if is_predicate_false(
+            &self.witness_map,
+            predicate,
+            self.pedantic_solving,
+            &opcode_location,
+        )? {
```

Comment:
Maybe `is_predicate_false` could be added to that `SolverContext` I suggested, resulting in a `ctx.is_predicate_false(predicate, &opcode_location)` which is a bit less busy.

Reply:
General response [above](https://github.com/noir-lang/noir/pull/6716#discussion_r1880892264)